### PR TITLE
Item component

### PR DESCRIPTION
**Pull request by leibole**

### CODE REVIEW

File: models/FeaturedItem.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            styleObjectColor: 'secondary'
+          - value: 'border-neutral'
+            label: 'Neutral color'
+            styleObjectColor: 'neutral'
+          - value: 'border-complementary'
+            label: 'Complementary color'
+            styleObjectColor: 'complementary'
+          - value: 'border-complementary-alt'
+            label: 'Complementary alt color'
+            styleObjectColor: 'complementaryAlt'
```
Comment (TomasBankauskas) [inline models/FeaturedItem.yaml:84]: `styleObjectColor` was removed use this link as example how to update the values -> https://github.com/stackbit/stackbit-components/blob/534876ade3b91f757435222271e67efa1213d066/models/HeroSection.yaml#L82

**Review by TomasBankauskas — CHANGES_REQUESTED** [review]

**Comment by JugglerX** [comment]
@leibole Hey a few things to make this process easier for me to review.

* Can you please add links to the Notion task and Figma design in the PR's description. 
* Can you please provide a brief summary of the purpose of this component?


I tested the component. It does not look like the components design is finished. I've attached a screenshot:

Screen Shot 2021-11-15 at 1 18 47 pm

**Note:** I added the `dist` folder so I could test your component in the Studio using github dependancies. It worked nicely. Sorry this creates a lot of junk into the PR. If @smnh approves adding the dist folder, we can move it to a seperate PR. 

Screen Shot 2021-11-15 at 1 36 42 pm

File: src/components/FeaturedItemsSection/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                sectionStyles.borderColor,
+                sectionStyles.borderRadius ? mapStyles({ borderRadius: sectionStyles.borderRadius }) : null,
+                sectionStyles.borderStyle ? mapStyles({ borderStyle: sectionStyles.borderStyle }) : null
+            )}
+            style={{
+                borderWidth: `${sectionBorderWidth}px`
+            }}
+            data-sb-field-path={props.annotationPrefix}
+        >
+            <div
```
Comment (JugglerX) [inline src/components/FeaturedItemsSection/index.tsx:33]: I wont comment on the main component until the design is finished. I tried a few of the different options in the Studio like changing the alignment and justify and they did not seem to align properly.

File: src/components/FeaturedItemsSection/featured-items-section.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                </>
+            )
+        }
+    }
+};
+
+const Template = (args) => <FeaturedItemsSection {...args} />;
+
+const args = {
+    type: 'FeaturedItemsSection',
```
Comment (JugglerX) [inline src/components/FeaturedItemsSection/featured-items-section.stories.tsx:60]: args values should match the defaults from the model `models/FeaturedItemsSection.yaml`

File: models/Item.yaml [hunk 4]
```diff
@@ -0,0 +1,132 @@
+type: object
```
Comment (JugglerX) [inline models/Item.yaml:1]: @TomasBankauskas Can you please review the model, I'm not really up to date with the latest model feature set.

Should it be called `ItemBlock`? I see we now have models called `Link.yaml` and `Social.yaml` so either we are no longer following this naming convention, or this model is not a Block.

If we are no longer following the naming convention, we need to remove Block from all the old models.

If we are still following the naming convention, can we get a clear definition of when to name something with `thingBlock` or `thing`

File: models/FeaturedItemsSection.yaml [hunk 5]
```diff
@@ -0,0 +1,121 @@
+type: object
```
Comment (JugglerX) [inline models/FeaturedItemsSection.yaml:1]: @TomasBankauskas Can you please review models, I'm not up to date with the latest model values, styles etc.

File: models/Item.yaml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+name: Item
+label: Item
+labelField: title
+fieldGroups:
+  - name: styles
+    label: Styles
+  - name: settings
+    label: Settings
+fields:
+  - type: enum
```
Comment (JugglerX) [inline models/Item.yaml:11]: Missing an `elementId` field. This allows every component to have a unique element id used as a fallback for css styling.

```
export default function HeroSection(props) {
    const cssId = props.elementId || null;
    
    return (
        <div
            id={cssId}
            className={classNames(
```

File: models/Item.yaml [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      type: model
+      models:
+        - Button
+        - Link
+    default:
+      - type: Button
+        label: Get Started
+        url: '/'
+        style: primary
+        elementId: hero-main-button
```
Comment (JugglerX) [inline models/Item.yaml:55]: delete elementId default

File: models/FeaturedItemsSection.yaml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      type: model
+      models:
+        - Button
+        - Link
+    default:
+      - type: Button
+        label: Get Started
+        url: '/'
+        style: primary
+        elementId: hero-main-button
```
Comment (JugglerX) [inline models/FeaturedItemsSection.yaml:50]: delete elementId default

File: models/FeaturedItemsSection.yaml [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  - Section
+groups:
+  - sectionComponent
+fieldGroups:
+  - name: styles
+    label: Styles
+  - name: settings
+    label: Settings
+fields:
+  - type: string
```
Comment (JugglerX) [inline models/FeaturedItemsSection.yaml:15]: Missing `elementId` field

File: src/components/Item/index.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+import { mapStylesToClassNames as mapStyles } from '../../utils/map-styles-to-class-names';
+
+export default function Item(props) {
+    const sectionStyles = props.styles?.self || {};
+
+    const sectionBorderWidth = sectionStyles.borderWidth ? sectionStyles.borderWidth : 0;
+    return (
+        <div
+            className={classNames(
```
Comment (JugglerX) [inline src/components/Item/index.tsx:15]: missing elementId and classes. I commented on this in more detail on the FeaturedItemSection component

File: src/components/Item/item.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+                </>
+            )
+        }
+    }
+};
+
+const Template = (args) => <Item {...args} />;
+
+const args = {
+    type: 'Item',
```
Comment (JugglerX) [inline src/components/Item/item.stories.tsx:60]: args should use the same values as the model default values

**Review by JugglerX — CHANGES_REQUESTED** [review]

**Comment by leibole** [comment]
@JugglerX These are the notion docs:
https://www.notion.so/stackbit/stackbit-components-Featured-Items-section-component-accfa01db7e447b98597afe611e2d060
https://www.notion.so/stackbit/stackbit-components-Item-component-9ceeb2cdf07d46df8f0e75d3856e766b

As you can see in the notion docs, there's still no design for these, which why I implemented something very basic.

File: models/FeaturedItemsSection.yaml (hunk 9 above)
Comment (leibole) [reply models/FeaturedItemsSection.yaml:15]: I think it's inherited from the section yaml.

Comment (JugglerX) [reply models/FeaturedItemsSection.yaml:15]: Yes I think you are right. No need to add it then.

**Review by TomasBankauskas — APPROVED** [review]

**Pull request merged**